aspnetcore/fundamentals/minimal-apis/10.0-samples/MinApiIproblemDetailsService/Program.cs

```diff
@@ -0,0 +1,58 @@
+// <snippet_register_IProblemDetailsService_implementation>
+using System.ComponentModel.DataAnnotations;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddProblemDetails(options =>
+{
+    options.CustomizeProblemDetails = context =>
+    {
+        if (context.ProblemDetails.Status == 400)
+        {
+            context.ProblemDetails.Title = "Validation error occurred";
+            context.ProblemDetails.Extensions["support"] = "Contact support@example.com";
+            context.ProblemDetails.Extensions["traceId"] = Guid.NewGuid().ToString();
+        }
+    };
+});
+// </snippet_register_IProblemDetailsService_implementation>
+
+var app = builder.Build();
+
+app.UseHttpsRedirection();
+
+// Define endpoints
+app.MapGet("/", () => "Hello! Use /products endpoint to test validation with IProblemDetailsService");
+
+// Products endpoint that demonstrates validation
+app.MapPost("/products", (Product product) => 
+{
+    // If validation passes (handled automatically based on DataAnnotations)
+    return Results.Ok(product);
+})
+.Accepts<Product>("application/json")
+.Produces<Product>(200)
+.ProducesProblem(400);
```

Comment:
Use `TypedResults` and the extension methods aren't needed.
```suggestion
app.MapPost("/products", (Product product) => 
{
    // If validation passes (handled automatically based on DataAnnotations)
    return TypedResults.Ok(product);
});
```